### PR TITLE
chore: update dependencies

### PR DESCRIPTION
**Pull request by vmx**
The code needs some slight changes as the linting rules got stricter.

### CODE REVIEW

File: test/index.spec.js [hunk 1]
```diff
@@ -345,7 +345,7 @@ describe('CID', () => {
       const hash = await multihashing(Buffer.from(`TEST${Date.now()}`), 'sha2-256')
       const cid = new CID(1, 'dag-pb', hash)
       expect(cid.buffer).to.equal(cid.buffer)
-      expect(Object.hasOwnProperty('buffer')).to.be.false()
+      expect(Object.prototype.hasOwnProperty.call(Object, 'buffer')).to.be.false()
```
Comment (alanshaw) [inline test/index.spec.js]: I'm not sure what this line is (or was) trying to test...

Currently I think it's testing that the `Object` constructor doesn't have a `buffer` property...

File: test/index.spec.js [hunk 2]
```diff
@@ -361,7 +361,7 @@ describe('CID', () => {
 
       // it cached!
       expect(cid.string).to.equal(base32String)
-      expect(Object.hasOwnProperty('_string')).to.be.false()
+      expect(Object.prototype.hasOwnProperty.call(Object, '_string')).to.be.false()
```
Comment (alanshaw) [inline test/index.spec.js]: Same here, I don't understand the purpose of this assertion. `_string` is not used in the code ever.

**Comment by vmx** [comment]
@alanshaw I also wasn't sure what this code was supposed to test, so I asked the original author @olizilla. I've added a comment explaining it.

**Comment by olizilla** [comment]
The tests were incorrect, the intention was that the `string` and `buffer` properties where internal implementation details, and would be lazily set at different times depending on how the instance was initiated so should not be trivially exposed or relied upon by the caller.

I should have just called them `_buffer` and `_string` to be more in line with old idomatic expectations.

**Pull request merged**